specification/cognitiveservices/data-plane/EntitySearch/v1.0/EntitySearch.json

```diff
@@ -0,0 +1,935 @@
+          {
+            "name": "ResponseFormat",
+            "in": "query",
+            "description": "The media type to use for the response. The following are the possible case-insensitive values: JSON, JSONLD. The default is JSON. If you specify JSONLD, the response body includes JSON-LD objects that contain the search results.",
+            "required": false,
+            "type": "array",
+            "items": {
+              "type": "string",
+              "enum": [
+                "Json",
+                "JsonLd"
+              ],
+              "x-ms-enum": {
+                "name": "ResponseFormat",
+                "modelAsString": false
+              }
+            },
+            "collectionFormat": "csv"
+          },
+          {
+            "name": "SafeSearch",
+            "in": "query",
+            "description": "A filter used to filter adult content. Off: Return webpages with adult text, images, or videos. Moderate: Return webpages with adult text, but not adult images or videos. Strict: Do not return webpages with adult text, images, or videos. The default is Moderate. If the request comes from a market that Bing's adult policy requires that safeSearch is set to Strict, Bing ignores the safeSearch value and uses Strict. If you use the site: query operator, there is the chance that the response may contain adult content regardless of what the safeSearch query parameter is set to. Use site: only if you are aware of the content on the site and your scenario supports the possibility of adult content.",
+            "required": false,
+            "type": "string",
+            "enum": [
+              "Off",
+              "Moderate",
+              "Strict"
+            ],
+            "x-ms-enum": {
+              "name": "SafeSearch",
+              "modelAsString": false
+            }
+          },
+          {
+            "name": "SetLang",
+            "in": "query",
+            "description": "The language to use for user interface strings. Specify the language using the ISO 639-1 2-letter language code. For example, the language code for English is EN. The default is EN (English). Although optional, you should always specify the language. Typically, you set setLang to the same language specified by mkt unless the user wants the user interface strings displayed in a different language. This parameter and the Accept-Language header are mutually exclusive; do not specify both. A user interface string is a string that's used as a label in a user interface. There are few user interface strings in the JSON response objects. Also, any links to Bing.com properties in the response objects apply the specified language.",
+            "required": false,
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success.",
+            "schema": {
+              "$ref": "#/definitions/SearchResponse"
+            }
+          },
+          "400": {
```

Comment:
Unfortunately, AutoRest will treat all modeled status codes as *successful* responses, while you probably want an exception to be raised in such case. Please refer to [this](https://github.com/Azure/azure-rest-api-specs/blob/current/documentation/creating-swagger.md#negative-responses) to learn more. In short: Create a `default` response of type `#/definitions/ErrorResponse`, I recommend making its description a pseudo table (Markdown, if you wish) containing the per-status-code descriptions you currently have 🙂 

Supporting a better modeling of error responses is high on our priority list.

Reply:
The default swagger generator client also threw an exception on an error response. I'm not sure why, exception driven development seems silly to me. Doesn't it make more sense to check the error response type? Instead of some random exception we can actually provide well-structured explanations of what happened. Thoughts?

Reply:
That does make sense, but is unfortunately not possible yet! If you generated the client like this right now, the method's return type would be `object` (since that's the lowest common base class of all your responses), and you would get an instance of the corresponding response type at runtime (i.e. an `ErrorResponse` object instead of an exception).

Reply:
well our success and error responses share a more common schema than object, but it's not a useful base. I don't really mind one over the other, if you're saying more people are building exception-throwing SDKs then it would be good to follow suit